components/Icon.tsx

```diff
@@ -0,0 +1,47 @@
+import clsx from "clsx";
+
+type Props = {
+  type: "home" | "search";
+  className?: string;
+  size?: string;
+};
+
+export default function Icon({ type, className, size }: Props) {
+  if (type === "home") {
+    return (
+      <>
+        <svg
+          className={clsx(size, className)}
```

Comment:
size is a string union like type. depending on which variation, you would conditionally add different classes.

medium would add h-7 w-7 maybe and small would add h-5 h-5 for example.

I'd use an object to store these variations where the key is part of the string union and the value is the classes to apply.

Reply:
for now, I'd just remove the size prop and send the width and height as part of the className for each icon

Reply:
Implemented sizes object, then defined sizeClass variable with ternary operator to handle size prop.